src/main/java/nl/hsac/fitnesse/fixture/util/HttpResponse.java

```diff
@@ -51,6 +51,7 @@ public String getRequest() {
      * @param aRequest the request to set
      */
     public void setRequest(String aRequest) {
+        aRequest = Environment.getInstance().getSoapSigningHelper().signSoapMessageIfNeeded(aRequest);
```

Comment:
I don't really like that the request is responsible for signing itself. Can this logic not be moved to XmlHttpTest?

Reply:
I looked at that, but It would intoduce a lot of duplication (or refactoring) around the use of templates, as these are translated to the request content in the Environment and the Environment eventually sends the request (as it holds the client)

Moving the signing to XmlHttpTest would impy that we process the template there as well. What do you think?

Reply:
I see your point. But I do believe the signing should not be a feature of HttpResponse. I see your point with the template handling in the environment. 
I'm not that happy anymore with the template handling (and calling the methods on HttpClient) being in the Environment . But there it is.

Maybe a custom subclass should have its own template handling, and just passing around the already signed result of a template. 
Another approach could be to add an extra (`Function<String,String>`) parameter to the methods in the environment that allows post processing of the template output. But this seems a bit over-the-top also.

Play around with it a bit. I think doing your own template handling is probably the most elegant solution

Reply:
I have put the WSSec methods in a separate xml http test extension now, that does its own template translation by overriding the necessary methods from http test. To do this, I had to change access for some fields on httptest from private to static. Could have used a getter, but it felt like overkill in this case. If you prefer using getters, i'll change them